types/lib/mapper.go

```diff
@@ -0,0 +1,276 @@
+package lib
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	wire "github.com/cosmos/cosmos-sdk/wire"
+)
+
+type GenericMapper struct {
+	key    sdk.StoreKey
+	cdc    *wire.Codec
+	prefix string
+}
+
+// ListMapper is a Mapper interface that provides list-like functions
+// It panics when the element type cannot be (un/)marshalled by the codec
+
+type ListMapper interface {
+	// ListMapper dosen't check if an index is in bounds
+	// The user should check Len() before doing any actions
+	Len(sdk.Context) uint64
+
+	Get(sdk.Context, uint64, interface{}) error
+
+	// Setting element out of range is harmful
+	// Use Push() instead of Set() to append a new element
+	Set(sdk.Context, uint64, interface{})
+
+	Delete(sdk.Context, uint64)
```

Comment:
Can we add a comment indicating how reindexing works after elements are deleted, and what happens when you delete an element out of bounds?